### PR TITLE
Remove doc generation features

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
As discussed, we will now call the commands (rustdoc-stripper) directly from the CI using the `generator.py` provided by `gir`. This is where the issue https://github.com/gtk-rs/gir/pull/1097 comes in handy (cc @MarijnS95).

Once this is one is merged, I'll send equivalents to gtk4 and gstreamer-rs.

### CODE REVIEW

File: gtk/README.md [hunk 1]
```diff
@@ -41,36 +41,14 @@ gtk = { git = "https://github.com/gtk-rs/gtk.git" }
...
-
-These features **rewrite the crate sources** so it's sufficient to enable them
-once. **Omitting them in the following `cargo` invocations will not undo their
-effects!**
-
 Generate the docs:
 
 ```shell
-> cargo doc --features v3_16
+> cargo doc --features "v3_16,dox"
```
Comment (bilelmoussaoui) [inline gtk/README.md:47]: dox only should be enough no?

**Comment by MarijnS95** [comment]
> This is where the issue gtk-rs/gir#1097 comes in handy (cc @MarijnS95).

We should see whether to invoke `rustdoc-stripper` from there too, to make this even more convenient (after all, we want to run `gir -m doc` and `rustdoc-stripper` over every crate that contains `Gir.toml` and is not a `sys` crate).

File: gtk/README.md (hunk 1 above)
Comment (GuillaumeGomez) [reply gtk/README.md:47]: True.

**Comment by GuillaumeGomez** [comment]
Yes, which is why I didn't approve your gir PR yet.

**Comment by GuillaumeGomez** [comment]
@bilelmoussaoui We will need to update the doc generation in the CI too so that it uses the new `generator.py` (once it's merged).

**Comment by bilelmoussaoui** [comment]
> @bilelmoussaoui We will need to update the doc generation in the CI too so that it uses the new `generator.py` (once it's merged).

sure, will take care of that once the new generator is merged :)

**Comment by MarijnS95** [comment]
> Yes, which is why I didn't approve your gir PR yet.

So that's the go-ahead to invoke `rustdoc-stripper` there? I'll try to get on it tonight then :)

**Review by sdroege — COMMENTED** [review]
Looks good to me but we should first get the generator PR merged (with having it run rustdoc-stripper etc) and switch over the CI to that at the same time. Otherwise the docs on the github pages will be incomplete for a while.

**Comment by GuillaumeGomez** [comment]
@sdroege Agreed. Now let's just wait on the gir PR and then everything will be "unlocked". :)

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ jobs:
       - uses: actions-rs/cargo@v1
```
Comment (MarijnS95) [inline .github/workflows/docs.yml:44]: @GuillaumeGomez You can substitute the invocations of `gir` and `rustdoc-stripper` above with `./generator.py --embed-docs` now :)

Comment (GuillaumeGomez) [reply .github/workflows/docs.yml:26]: Not working.

Comment (bilelmoussaoui) [reply .github/workflows/docs.yml:26]: It does, you need a bit more than calling the generator here; see what I did at https://github.com/gtk-rs/gtk4-rs/pull/313

Comment (MarijnS95) [reply .github/workflows/docs.yml:26]: @GuillaumeGomez Have you deleted `generator.py` and symlinked it to `./gir/generator.py`?

Comment (MarijnS95) [reply .github/workflows/docs.yml:27]: (If the error message were wrong, blame `argparse` :smirk:)

Comment (GuillaumeGomez) [reply .github/workflows/docs.yml:27]: Ah, I didn't!

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -22,29 +22,13 @@ jobs:
...
-      - run: ./gir/target/release/gir -c ./atk/Gir.toml -d ./gir-files --doc-target-path docs.md -m doc
-      - run: ./gir/target/release/gir -c ./gdk/Gir.toml -d ./gir-files --doc-target-path docs.md -m doc
-      - run: ./gir/target/release/gir -c ./gdk-pixbuf/Gir.toml -d ./gir-files --doc-target-path docs.md -m doc
-      - run: ./gir/target/release/gir -c ./gdkx11/Gir.toml -d ./gir-files --doc-target-path docs.md -m doc
-      - run: ./gir/target/release/gir -c ./gio/Gir.toml -d ./gir-files --doc-target-path docs.md -m doc
-      - run: ./gir/target/release/gir -c ./graphene/Gir.toml -d ./gir-files --doc-target-path docs.md -m doc
-      - run: ./gir/target/release/gir -c ./gtk/Gir.toml -d ./gir-files --doc-target-path docs.md -m doc
-      - run: ./gir/target/release/gir -c ./pango/Gir.toml -d ./gir-files --doc-target-path docs.md -m doc
-      - run: ./gir/target/release/gir -c ./pangocairo/Gir.toml -d ./gir-files --doc-target-path docs.md -m doc
+      - run: rustup component add rustfmt
```
Comment (MarijnS95) [inline .github/workflows/docs.yml:25]: You can also use `components: rustfmt` in `actions-rs/toolchain@v1`, bit cleaner IMO (cc @bilelmoussaoui, iirc it's the same command in gtk4-rs).

Comment (bilelmoussaoui) [reply .github/workflows/docs.yml:25]: @MarijnS95 ah, i didn't knew that was possible :p I will update those some other day when I need to fix something on the CI

**Comment by GuillaumeGomez** [comment]
Does anyone know what is the `env!` error? When compiling tests locally it works fine for me.

**Comment by GuillaumeGomez** [comment]
ping @sdroege @bilelmoussaoui

**Comment by bilelmoussaoui** [comment]
Couldn't it be related to the env variables requiring a build.rs to be used ?

**Comment by MarijnS95** [comment]
> Couldn't it be related to the env variables requiring a build.rs to be used ?

That's true, https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates:

> `OUT_DIR` — If the package has a build script, this is set to the folder where the build script should place its output. See below for more information. (Only set during compilation.)

**Comment by MarijnS95** [comment]
Looking at the code `gio/src/settings.rs` is only hoisting the build directory for some runtime intermediates used in the tests... Perhaps just use `/tmp`?

**Review by bilelmoussaoui — APPROVED** [review]

File: .github/workflows/docs.yml (hunk 3 above)
Comment (MarijnS95) [reply .github/workflows/docs.yml:25]: Ping @GuillaumeGomez ^^

File: gio/build.rs [hunk 4]
```diff
@@ -1,22 +1,3 @@
...
-}
-
-#[cfg(any(
-    all(feature = "embed-lgpl-docs", feature = "purge-lgpl-docs"),
-    not(any(feature = "embed-lgpl-docs", feature = "purge-lgpl-docs"))
-))]
-fn manage_docs() {}
+// This whole build.rs file is required because of the `OUT_DIR` environment variable
+// provided by cargo when a build file is present.
+fn main() {}
```
Comment (MarijnS95) [inline gio/build.rs:3]: That's not totally descriptive (might want to mention the tests need `OUT_DIR` to store a temporary file, and that environment variable is only defined when the crate has a `build.rs` script), but okay.

**Review by MarijnS95 — APPROVED** [review]
Nice cleanup, sad to see some CI warnings in unrelated, manual code though.

File: .github/workflows/docs.yml (hunk 3 above)
Comment (GuillaumeGomez) [reply .github/workflows/docs.yml:25]: @MarijnS95 Please send another PR for that. ;)

**Comment by GuillaumeGomez** [comment]
Gonna send a PR on gtk4 now if not already done.

**Comment by bilelmoussaoui** [comment]
> Gonna send a PR on gtk4 now if not already done.

already done

**Comment by GuillaumeGomez** [comment]
Then my job is done here. :)

**Pull request merged**